controllers/podchaos/containerkill/types.go

```diff
@@ -0,0 +1,190 @@
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	filteredPod, err := utils.GeneratePods(pods, podchaos.Spec.Mode, podchaos.Spec.Value)
+	if err != nil {
+		r.Log.Error(err, "fail to generate pods")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	g := errgroup.Group{}
+	haveSelected := false
+	haveKilled := false
+	for index := range filteredPod {
+		pod := &filteredPod[index]
+
+		for index := range pod.Status.ContainerStatuses {
+			containerName := pod.Status.ContainerStatuses[index].Name
+			containerID := pod.Status.ContainerStatuses[index].ContainerID
+
+			if containerName == podchaos.Spec.ContainerName {
+				haveSelected = true
+				err = r.KillContainer(ctx, pod, containerID)
+				if err != nil {
+					r.Log.Error(err, "failed to kill container")
+				} else {
+					haveKilled = true
+				}
+			}
+		}
+	}
+
+	if haveSelected == false {
+		err = errors.New("no container is selected")
+		r.Log.Error(err, "no container is selected")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if haveKilled == false {
+		err = errors.New("no container is killed")
+		r.Log.Error(err, "no container is killed")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if err := g.Wait(); err != nil {
+		return ctrl.Result{}, nil
+	}
+
+	return r.updatePodchaos(ctx, podchaos, pods, now)
+}
+
+// KillContainer kills container according to containerID
+// Use client in chaos-daemon
+func (r *Reconciler) KillContainer(ctx context.Context, pod *v1.Pod, containerID string) error {
+
+	r.Log.Info("try to kill container", "namespace", pod.Namespace, "podName", pod.Name, "containerID", containerID)
+
+	c, err := utils.CreateGrpcConnection(ctx, r.Client, pod)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	pbClient := pb.NewChaosDaemonClient(c)
+
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return fmt.Errorf("%s %s can't get the state of container", pod.Namespace, pod.Name)
+	}
+
+	if _, err = pbClient.ContainerKill(ctx, &pb.ContainerRequest{
+		ContainerId: containerID,
+	}); err != nil {
+		r.Log.Error(err, "kill container error", "namespace", pod.Namespace, "podName", pod.Name, "containerID", containerID)
+	}
+
+	return err
+}
+
+func (r *Reconciler) updatePodchaos(ctx context.Context, podchaos v1alpha1.PodChaos, pods []v1.Pod, now time.Time) (ctrl.Result, error) {
+	next, err := utils.NextTime(*podchaos.Spec.Scheduler, now)
+	if err != nil {
+		r.Log.Error(err, "failed to get next time")
+		return ctrl.Result{}, nil
+	}
+
+	podchaos.SetNextStart(*next)
+
+	podchaos.Status.Experiment.StartTime = &metav1.Time{
+		Time: now,
+	}
+	podchaos.Status.Experiment.EndTime = &metav1.Time{
+		Time: now,
+	}
+
+	podchaos.Status.Experiment.Pods = []v1alpha1.PodStatus{}
+	for _, pod := range pods {
+		ps := v1alpha1.PodStatus{
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
+			HostIP:    pod.Status.HostIP,
+			PodIP:     pod.Status.PodIP,
+			Action:    string(podchaos.Spec.Action),
+			Message:   containerKillActionMsg,
```

Comment:
Record which container was killed

Reply:
Because there is no parameter "container ID "passed, it is changed to record the "container name" here.